aspnetcore/mvc/advanced/custom-model-binding/sample/CustomModelBindingSample/Binders/AuthorEntityBinder.cs

```diff
@@ -0,0 +1,75 @@
+﻿using CustomModelBindingSample.Data;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CustomModelBindingSample.Binders
+{
+    public class AuthorEntityBinder : IModelBinder
+    {
+        private readonly AppDbContext _db;
+        public AuthorEntityBinder(AppDbContext db)
+        {
+            _db = db;
+        }
+        public Task BindModelAsync(ModelBindingContext bindingContext)
+        {
+            if (bindingContext == null)
+            {
+                throw new ArgumentNullException(nameof(bindingContext));
+            }
+
+            // specify a default argument name if none is set
+            var modelName = bindingContext.ModelName;
+            if(string.IsNullOrEmpty(modelName))
+            {
+                modelName = "authorId";
+            }
```

Comment:
If I don't have that, my provider doesn't work, since the model name is always empty (unless set by ModelBinderAttribute). So this method never uses the model binder correctly:
```
        [HttpGet("get/{authorId}")]
        public IActionResult GetById2(Author author)
        {
            // this example works with the AuthorEntityBinderProvider
            // you must remove the ModelBinder attribute from Author
            // for this provider to work
            return Ok(author);
        }
```
When `Author` has the attribute, it sets the 'Name' to 'Id' which also makes this not work. However, if I remove the attribute and add the default model name logic ("authorId") then it works. How would you implement it so that it binds to an expected "id" parameter?